orangecontrib/imageanalytics/widgets/owimagenetembedding.py

```diff
@@ -99,28 +107,33 @@ def set_data(self, data):
             return
 
         self.info_a.setText("Data with %d instances." % len(data))
-        self.data = data
-        # todo check for image attributes
-
-        atts = [a for a in data.domain.metas if
+        self.atts = [a for a in data.domain.metas if
                 a.attributes.get("type") == "image"]
-        if not atts:
+        if not self.atts:
             self.warning(text="Input data has no image attributes.")
             self.info_a.setText("Data (%d instances) without image "
                                 "attributes." % len(data))
             self.data = None
             return
-        self.file_att = atts[0]  # todo handle a set of image attributes
-        self.origin = self.file_att.attributes.get("origin", None) or \
-            self.data.attributes.get("origin", None)
-        self.commit()
+        self.img_cb.setModel(VariableListModel(self.atts))
+        self.img_cb.setCurrentIndex(self.img_attr)
+
+        self.data = data
+        self.img_attr_changed()
 
     def token_name_changed(self):
         self.profiler.set_token(self.token)
         if self.profiler.token and self.data:  # token is valid
             self.commit()
         self.set_info()
 
+    def img_attr_changed(self):
+        self.file_att = self.atts[self.img_attr]
```

Comment:
This raises an index error when switching between inputs with different number of image attributes:
```
--------------------------------------------------------------------------------
IndexError                                    Traceback (most recent call last):
  File "/Users/aleserjavec/workspace/orange3/Orange/canvas/scheme/widgetsscheme.py", line 699, in process_signals_for_widget
    handler(*args)
  File "/Users/aleserjavec/workspace/orange3-imageanalytics/orangecontrib/imageanalytics/widgets/owimagenetembedding.py", line 122, in set_data
    self.img_attr_changed()
  File "/Users/aleserjavec/workspace/orange3-imageanalytics/orangecontrib/imageanalytics/widgets/owimagenetembedding.py", line 131, in img_attr_changed
    self.file_att = self.atts[self.img_attr]
IndexError: list index out of range
```